### PR TITLE
Add unit test for resolveConflicts function

### LINKED ISSUE #5079: resolveConflicts is not covered by unit tests
**Describe the bug**
resolveConflicts is not fully covered by unit tests
https://github.com/cortexproject/cortex/pull/5061#pullrequestreview-1230645707

### PR DESCRIPTION
**Pull request by dogukanteber**
**What this PR does**:
This PR adds unit test to `resolveConflicts` function

**Which issue(s) this PR fixes**:
Fixes #5079 

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ring/model_test.go [hunk 1]
```diff
@@ -600,3 +601,86 @@ func TestDesc_FindDifference(t *testing.T) {
...
+		},
+		{
+			name: "Conflict resolution with LEFT state",
+			args: map[string]InstanceDesc{
+				"ing1": {State: ACTIVE, Tokens: []uint32{1, 2, 3}},
+				"ing2": {State: LEFT, Tokens: []uint32{1, 2, 3, 4}},
+			},
+			want: map[string]InstanceDesc{
+				"ing1": {State: ACTIVE, Tokens: []uint32{1, 2, 3}},
+				"ing2": {State: LEFT, Tokens: []uint32{}},
```
Comment (friedrichg) [inline pkg/ring/model_test.go:635]:
```suggestion
				"ing2": {State: LEFT, Tokens: nil},
```

File: pkg/ring/model_test.go [hunk 2]
```diff
@@ -600,3 +601,86 @@ func TestDesc_FindDifference(t *testing.T) {
...
+			},
+		},
+		{
+			name: "Conflict resolution with LEAVING state",
+			args: map[string]InstanceDesc{
+				"ing1": {State: LEAVING, Tokens: []uint32{1, 2, 3}},
+				"ing2": {State: PENDING, Tokens: []uint32{1, 2, 3, 4}},
+			},
+			want: map[string]InstanceDesc{
+				"ing1": {State: LEAVING, Tokens: []uint32{}},
```
Comment (friedrichg) [inline pkg/ring/model_test.go:645]:
```suggestion
				"ing1": {State: LEAVING, Tokens: nil},
```

File: pkg/ring/model_test.go [hunk 3]
```diff
@@ -600,3 +601,86 @@ func TestDesc_FindDifference(t *testing.T) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resolveConflicts(tt.args)
+			for key, actualInstance := range tt.args {
+				expectedInstance := tt.want[key]
+				if actualInstance.State != expectedInstance.State || !reflect.DeepEqual(actualInstance.Tokens, expectedInstance.Tokens) {
+					if len(actualInstance.Tokens) == len(expectedInstance.Tokens) && len(actualInstance.Tokens) != 0 {
+						t.Errorf("resolveConflicts() for key %s = %v, want %v", key, actualInstance, expectedInstance)
+					}
+				}
```
Comment (friedrichg) [inline pkg/ring/model_test.go:682]:
```suggestion
				assert.Equal(t, actualInstance, tt.want[key])
```

File: pkg/ring/model_test.go [hunk 4]
```diff
@@ -600,3 +601,86 @@ func TestDesc_FindDifference(t *testing.T) {
...
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resolveConflicts(tt.args)
+			for key, actualInstance := range tt.args {
+				expectedInstance := tt.want[key]
+				if actualInstance.State != expectedInstance.State || !reflect.DeepEqual(actualInstance.Tokens, expectedInstance.Tokens) {
+					if len(actualInstance.Tokens) == len(expectedInstance.Tokens) && len(actualInstance.Tokens) != 0 {
```
[the remark this answers is not in the document]
Comment (friedrichg) [reply pkg/ring/model_test.go:679]: DeepEqual was telling you the right thing.
`nil` is different from empty array `[]uint32{}`

assertEqual tells you this more clearly. testify is a great library

**Review by friedrichg — CHANGES_REQUESTED** [review]

**Review by friedrichg — APPROVED** [review]

**Review by alvinlin123 — APPROVED** [review]

**Pull request merged**